src/pages/index.js

```diff
@@ -0,0 +1,321 @@
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+  console.log("Form submitted");
+  api
+    .editUserInfo({ name: nameInput.value, about: descriptionInput.value })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+      evt.target.reset();
+      disableButton(
+        editFormElement.querySelector(".modal__submit-btn"),
+        settings
+      );
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
+}
+
+function handleAddCardFormSubmit(evt) {
+  console.log("Form submitted");
+  evt.preventDefault();
+  api
+    .addCard({ name: cardNameInput.value, link: cardLinkInput.value })
+    .then((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardList.prepend(cardElement);
+      closeModal(cardModal);
+      evt.target.reset();
+      disableButton(cardSubmitButton, settings);
+    })
+    .catch(console.error);
+}
+profileEditButton.addEventListener("click", () => {
+  nameInput.value = profileName.textContent;
+  descriptionInput.value = profileDescription.textContent;
+  resetValidation(editFormElement, [nameInput, descriptionInput], settings);
+  openModal(editModal);
+});
+
+closeModalButton.addEventListener("click", () => {
+  closeModal(editModal);
+});
+
+cardModalOpenButton.addEventListener("click", () => {
+  const inputList = [cardNameInput, cardLinkInput];
+
+  openModal(cardModal);
+});
+cardModalCloseButton.addEventListener("click", () => {
+  closeModal(cardModal);
+});
+
+deleteModalCancelButton.addEventListener("click", () => {
+  closeModal(deleteModal);
+});
+
+avatarModalCloseButton.addEventListener("click", () => {
+  closeModal(avatarModal);
+});
+
+avatarModalOpenButton.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+
+editFormElement.addEventListener("submit", handleEditFormSubmit);
+cardForm.addEventListener("submit", handleAddCardFormSubmit);
+
+const previewModalCloseButton = previewModal.querySelector(
+  ".modal__close-btn_preview"
+);
+avatarForm.addEventListener("submit", handleAvatarSubmit);
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  console.log("Avatar URL being submitted:", avatarInput.value); // Add this line
+
+  api
+    .editAvatarInfo({ avatar: avatarInput.value })
+    .then((data) => {
+      console.log("Response from server:", data); // Add this line
+
+      profileAvatar.src = data.avatar;
+      closeModal(avatarModal);
+      evt.target.reset();
+      disableButton(avatarSubmitButton, settings);
+    })
+    .catch((error) => {
+      console.error("Error updating avatar:", error); // Add this line
+    });
+}
+let selectedCard;
+let selectedCardId;
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
```

Comment:
* According to the task, you need to change submit buttons text (`Saving…`) while waiting for the server response. And the user should see it.

And when you delete, it should be `Deleting...`